src/vmm/src/device_manager/legacy.rs

```diff
@@ -35,9 +35,9 @@ pub enum LegacyDeviceError {
 pub struct PortIODeviceManager {
     pub io_bus: crate::devices::Bus,
     // BusDevice::Serial
-    pub stdio_serial: Arc<Mutex<BusDevice>>,
+    pub stdio_serial: BusDevice,
```

Comment:
Ah, lol, so we did face the problem of not being able to share devices properly before already! We just relied on the awkward matches apparently. Well, now that we have this refactor, we can clean this up and just store a `Arc<Mutex<Serial>>` here and a `Arc<Mutex<I8042Device>>` below :)